### DISCUSSION TITLE
Is the default shell to run `%post` section `sh` or `bash` on Ubuntu 22.04?

### DISCUSSION THREAD

**Discussion by Heng-Zhou** [sylabs/singularity#2528] [Q&A] [answered]
[The document](https://docs.sylabs.io/guides/latest/user-guide/definition_files.html#post) of `%post` section says 
![image](https://github.com/sylabs/singularity/assets/111557423/dd018b0c-86b8-42b0-9b50-418ce0ca0510)
Which then is the default shell to run `%post` section on Ubuntu 22.04?

**Comment by dtrudg** [comment, marked as answer]
`/bin/sh` is called by default. On many distributions this is a symlink to bash. However, on Ubuntu it is `dash`, which is a POSIX `sh` shell.

Generally, it is safest to write `%post` etc. assuming they will run on a POSIX `sh` compatible shell. If you need to use non-sh syntax, specify a custom shell with `-c` as documented - https://docs.sylabs.io/guides/latest/user-guide/definition_files.html#post

**Comment by Heng-Zhou** [comment]
Thanks for the reply.
